README.md

```diff
@@ -59,14 +66,54 @@ This library has been tested with modem firmware version `1.3.4` but might work
 When this library starts to require a newer version, then that will be seen as a breaking change.
 But it's easy to miss something, so this is a 'best effort' guarantee only.
 
-### Nonsecure
+### Memory
+
+The model library from Nordic needs some memory for its state and buffers. You need to reserve some memory in your memory.x file for the modem:
+
+```ld
+MEMORY
+{
+    FLASH : ORIGIN = 0x00000000, LENGTH = 1024K
+    MODEM : ORIGIN = 0x20000000, LENGTH = 32K
+    RAM   : ORIGIN = 0x20008000, LENGTH = 224K
+}
+```
+
+### Secure and nonsecure operation
+
+The library can be used in secure and non-secure contexts. Some additional initialization is necessary for the secure context because the underlying libmodem C library by Nordic expects access to nonsecure memory and resources. If you do not use the memory layout defined above, you need to adapt the addresses below. 
+
```

Comment:
You do mention this code sample is only for when running in the secure context. However, when you only skim the text, that's easily missed.

```suggestion
The library can be used in secure and non-secure contexts. Some additional initialization is necessary for the secure context because the underlying libmodem C library by Nordic expects access to nonsecure memory and resources. If you do not use the memory layout defined above, you need to adapt the addresses below. 

For running in the secure context:
```

Reply:
Nice though to elaborate more on this in the readme!